org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java

```diff
@@ -125,6 +126,17 @@ public CompletionList doComplete(DOMDocument xmlDocument, Position position, Sha
 						return completionResponse;
 					}
 					break;
+				case DTDStartDoctypeTag:
+					DTDDeclParameter systemId = xmlDocument.getDoctype().getSystemIdNode();
+					if (systemId == null) {
+						break;
+					}
+					if (systemId.getStart() <= offset && offset <= systemId.getEnd()) {

```

Comment:
I think you could replace with `DOMNode.isIncluded(systemId, offset)`

Reply:
fixed